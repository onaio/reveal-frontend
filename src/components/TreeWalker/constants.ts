export const ACTIVE = 'Active';
export const FEATURE = 'Feature';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_CHILDREN = 'Could not load children';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
export const NOT_AVAILABLE = 'N/A';
