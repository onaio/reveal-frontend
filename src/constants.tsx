// strings
export const MAP = 'map';
export type MAP = typeof MAP;
export const LOCATION = 'Location';
export type LOCATION = typeof LOCATION;
export const PROVINCE = 'Province';
export type PROVINCE = typeof PROVINCE;
export const HOME = 'Home';
export type HOME = typeof HOME;
export const FOCUS_INVESTIGATION = 'Focus Investigation';
export type FOCUS_INVESTIGATION = typeof FOCUS_INVESTIGATION;
export const ACTIVE_FOCUS_INVESTIGATION = 'Active Focus Investigations';
export type ACTIVE_FOCUS_INVESTIGATION = typeof ACTIVE_FOCUS_INVESTIGATION;
export const FOCUS_AREA_INFO = 'Focus Area Information';
export type FOCUS_AREA_INFO = typeof FOCUS_AREA_INFO;
export const FOCUS_INVESTIGATIONS = 'Focus Investigations';
export type FOCUS_INVESTIGATIONS = typeof FOCUS_INVESTIGATIONS;
export const MAP_ID = 'map-1';
export type MAP_ID = typeof MAP_ID;
export const GEOJSON = 'geojson';
export type GEOJSON = typeof GEOJSON;
export const STRINGIFIED_GEOJSON = 'stringified-geojson';
export type STRINGIFIED_GEOJSON = typeof STRINGIFIED_GEOJSON;
export const FOCUS_AREA_HEADER = 'Focus Area';
export type FOCUS_AREA_HEADER = typeof FOCUS_AREA_HEADER;
export const REASON_HEADER = 'Reason';
export type REASON_HEADER = typeof REASON_HEADER;
export const STATUS_HEADER = 'Status';
export type STATUS_HEADER = typeof STATUS_HEADER;
export const CASE_NOTIF_DATE_HEADER = 'Case Notif. Date';
export type CASE_NOTIF_DATE_HEADER = typeof CASE_NOTIF_DATE_HEADER;
export const CASE_CLASSIFICATION_HEADER = 'Case Class.';
export type CASE_CLASSIFICATION_HEADER = typeof CASE_CLASSIFICATION_HEADER;
export const DEFINITIONS = 'Definitions';
export type DEFINITIONS = typeof DEFINITIONS;
export const IN = 'in';
export type IN = typeof IN;
export const DISTRICT = 'District';
export type DISTRICT = typeof DISTRICT;
export const CANTON = 'Canton';
export type CANTON = typeof CANTON;
export const FI_REASON = 'FI Reason';
export type FI_REASON = typeof FI_REASON;
export const FI_STATUS = 'FI Status';
export type FI_STATUS = typeof FI_STATUS;
export const ACTIVE_INVESTIGATION = 'Active Investigation';
export type ACTIVE_INVESTIGATION = typeof ACTIVE_INVESTIGATION;
export const COMPLETE = 'Complete';
export type COMPLETE = typeof COMPLETE;
export const NO = 'No';
export type NO = typeof NO;
export const RESPONSE = 'Response';
export type RESPONSE = typeof RESPONSE;
export const MEASURE = 'Measure';
export type MEASURE = typeof MEASURE;
export const OF = 'of';
export type OF = typeof OF;
export const MARK_AS_COMPLETE = 'Mark as complete';
export type MARK_AS_COMPLETE = typeof MARK_AS_COMPLETE;
export const TARGET = 'Target';
export type TARGET = typeof TARGET;
export const POLYGON = 'Polygon';
export type POLYGON = typeof POLYGON;
export const MULTI_POLYGON = 'MultiPolygon';
export type MULTI_POLYGON = typeof MULTI_POLYGON;
export const POINT = 'Point';
export type POINT = typeof POINT;
export const FEATURE = 'Feature';
export type FEATURE = typeof FEATURE;
export const FEATURE_COLLECTION = 'FeatureCollection';
export type FEATURE_COLLECTION = typeof FEATURE_COLLECTION;
export const MAIN_PLAN = 'main-plan-layer';
export type MAIN_PLAN = typeof MAIN_PLAN;
export const APP = 'APP';
export type APP = typeof APP;
export const NO_GEOMETRIES_RESPONSE = 'Goals have no Geometries';
export type NO_GEOMETRIES_RESPONSE = typeof NO_GEOMETRIES_RESPONSE;

// internal urls
export const LOGIN_URL = '/login';
export type LOGIN_URL = typeof LOGIN_URL;
export const LOGOUT_URL = '/logout';
export type LOGOUT_URL = typeof LOGOUT_URL;
export const HOME_URL = '/';
export type HOME_URL = typeof HOME_URL;
export const IRS_URL = '/irs';
export type IRS_URL = typeof IRS_URL;
export const FI_URL = '/focus-investigation';
export type FI_URL = typeof FI_URL;
export const FI_SINGLE_URL = '/focus-investigation/view';
export type FI_SINGLE_URL = typeof FI_SINGLE_URL;
export const FI_SINGLE_MAP_URL = '/focus-investigation/map';
export type FI_SINGLE_MAP_URL = typeof FI_SINGLE_MAP_URL;
export const FI_HISTORICAL_URL = '/focus-investigation/historical';
export type FI_HISTORICAL_URL = typeof FI_HISTORICAL_URL;

// colors
export const GREEN = 'Green';
export type GREEN = typeof GREEN;
export const YELLOW = 'Yellow';
export type YELLOW = typeof YELLOW;
export const ORANGE = 'Orange';
export type ORANGE = typeof ORANGE;
export const RED = 'Red';
export type RED = typeof RED;

// task action codes
export const RACD_REGISTER_FAMILY_CODE = 'RACD Register Family';
export type RACD_REGISTER_FAMILY_CODE = typeof RACD_REGISTER_FAMILY_CODE;
export const CASE_CONFIRMATION_CODE = 'Case Confirmation';
export type CASE_CONFIRMATION_CODE = typeof CASE_CONFIRMATION_CODE;
export const MOSQUITO_COLLECTION_CODE = 'Mosquito Collection';
export type MOSQUITO_COLLECTION_CODE = typeof MOSQUITO_COLLECTION_CODE;
export const BEDNET_DISTRIBUTION_CODE = 'Bednet Distribution';
export type BEDNET_DISTRIBUTION_CODE = typeof BEDNET_DISTRIBUTION_CODE;
export const LARVAL_DIPPING_CODE = 'Larval Dipping';
export type LARVAL_DIPPING_CODE = typeof LARVAL_DIPPING_CODE;
export const BLOOD_SCREENING_CODE = 'Blood Screening';
export type BLOOD_SCREENING_CODE = typeof BLOOD_SCREENING_CODE;
export const IRS_CODE = 'IRS';
export type IRS = typeof IRS_CODE;
