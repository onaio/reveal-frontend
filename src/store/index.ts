import { store } from '@opensrp/store';

export default store;
