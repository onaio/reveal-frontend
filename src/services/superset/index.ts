import reducerRegistry from '@onaio/redux-reducer-registry';
import superset, { SupersetConnectorConfig } from '@onaio/superset-connector';
import { OPENSRP_OAUTH_STATE, SUPERSET_API_BASE, SUPERSET_API_ENDPOINT } from '../../configs/env';
import { FlexObject } from '../../helpers/utils';
import store from '../../store';
import supersetReducer, {
  authorizeSuperset,
  isAuthorized,
  reducerName as supersetReducerName,
} from '../../store/ducks/superset';
import { getAccessToken, getOauthProviderState } from '../../store/selectors';

// /** register the superset reducer */
reducerRegistry.register(supersetReducerName, supersetReducer);

/** middleware for fetching from Superset */
export const fetchMiddleware = (res: { [key: string]: any }) => {
  return res;
};

/** callback for fetching from Superset */
export const fetchCallback = (parsedResponse: Array<{ [key: string]: any }>) => {
  const sliceData = superset.processData(parsedResponse);
  return sliceData;
};

/** this function completes the authZ process */
export const completeAuthZ = async (result: { [key: string]: any }) => {
  if (result.status === 200) {
    store.dispatch(authorizeSuperset(true));
  } else {
    store.dispatch(authorizeSuperset(false));
  }
};

/** generic async function for fetching from superset */
const supersetFetch = async (
  sliceId: string,
  params: FlexObject | null = null,
  callback: typeof fetchCallback = fetchCallback,
  middleware: typeof fetchMiddleware = fetchMiddleware
) => {
  const config: SupersetConnectorConfig = {
    base: SUPERSET_API_BASE,
    endpoint: SUPERSET_API_ENDPOINT,
    extraPath: sliceId,
    /** the provider should be the name of the provider
     * to enable the use of multiple providers, we store the provider name in the
     * `state` parameter so that we can use it here
     */
    provider: getOauthProviderState(store.getState()) || OPENSRP_OAUTH_STATE,
    token: '',
  };

  const accessToken = getAccessToken(store.getState());
  if (accessToken) {
    config.token = accessToken;
  }

  if (params) {
    config.params = `form_data=${JSON.stringify(params)}`;
  }

  const isSupersetAuthorized = isAuthorized(store.getState());

  if (isSupersetAuthorized === true) {
    return superset.api.doFetch(config, middleware).then(callback);
  } else {
    return superset.authZ(config, (result: { [key: string]: any }) => {
      return completeAuthZ(result)
        .then(() => {
          const isSupersetAuthorizedYet = isAuthorized(store.getState());
          if (isSupersetAuthorizedYet === true) {
            return superset.api.doFetch(config, middleware).then(callback);
          }
        })
        .catch(() => {
          return false;
        });
    });
  }
};

export default supersetFetch;
