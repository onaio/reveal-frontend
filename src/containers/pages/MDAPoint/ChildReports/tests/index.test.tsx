import reducerRegistry from '@onaio/redux-reducer-registry';
import superset from '@onaio/superset-connector';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import ConnectedChildReports, { ChildReportList } from '..';
import { MDA_POINT_CHILD_REPORT_URL } from '../../../../../constants';
import store from '../../../../../store';
import GenericJurisdictionsReducer, {
  fetchGenericJurisdictions,
  reducerName as GenericJurisdictionsReducerName,
} from '../../../../../store/ducks/generic/jurisdictions';
import MDAPointChildReportReducer, {
  FetchMDAPointChildReportAction,
  reducerName as MDAPointChildReportReducerName,
  removeMDAPointChildReports,
} from '../../../../../store/ducks/generic/MDAChildReport';
import GenericPlanreducer, {
  genericFetchPlans,
  GenericPlan,
  reducerName as genericReducerName,
} from '../../../../../store/ducks/generic/plans';
import * as fixtures from '../../../../../store/ducks/generic/tests/fixtures';
import { MDAPointJurisdictionsJSON } from '../../jurisdictionsReport/tests/fixtures';

jest.mock('../../../../../configs/env');
/* tslint:disable-next-line no-var-requires */
const fetch = require('jest-fetch-mock');

const history = createBrowserHistory();

reducerRegistry.register(MDAPointChildReportReducerName, MDAPointChildReportReducer);
reducerRegistry.register(GenericJurisdictionsReducerName, GenericJurisdictionsReducer);
reducerRegistry.register(genericReducerName, GenericPlanreducer);

const jurisdictionData = superset.processData(MDAPointJurisdictionsJSON) || [];
store.dispatch(fetchGenericJurisdictions('esw-jurisdictions', jurisdictionData));
store.dispatch(genericFetchPlans(fixtures.MDAPointPlans as GenericPlan[]));

const props = {
  history,
  location: {
    hash: '',
    pathname: MDA_POINT_CHILD_REPORT_URL,
    search: '',
    state: undefined,
  },
  match: {
    isExact: true,
    params: {
      jurisdictionId: '3951',
      planId: '40357eff-81b6-4e32-bd3d-484019689f7c',
    },
    path: `${MDA_POINT_CHILD_REPORT_URL}`,
    url: `${MDA_POINT_CHILD_REPORT_URL}`,
  },
};

describe('components/MDA Reports/MDAPlansList', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    shallow(
      <Router history={history}>
        <ChildReportList {...props} />
      </Router>
    );
  });

  it('should render school reports correctly', () => {
    store.dispatch(FetchMDAPointChildReportAction(fixtures.MDAPointChildReportData));

    const tableData = [
      ['annonymous user', '2345', 'No', 'Yes', 'No', 'No', 'No', 0, 0],
      ['known man', '843', 'No', 'No', 'Yes', 'Yes', 'No', 0, 0],
    ];

    fetch.mockResponseOnce(fixtures.MDAPointChildReportData);

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedChildReports {...props} />
        </Router>
      </Provider>
    );

    expect(wrapper.find('h3.page-title').text()).toEqual('MDA Point Child Report: Akros_1');
    expect(toJson(wrapper.find('BreadcrumbItem li'))).toMatchSnapshot('breadcrumbs');

    // correct table data is passed
    expect(wrapper.find('ChildSupersetDataTable').props().data).toEqual(tableData);
    expect(wrapper.find('ListView').props().data).toEqual(tableData);

    expect(toJson(wrapper.find('ListView table'))).toMatchSnapshot('table');
    // two reports rendered
    expect(wrapper.find('.listview-tbody tr').length).toEqual(2);

    // clear store
    store.dispatch(removeMDAPointChildReports());
    wrapper.update();
    expect(wrapper.find('ChildSupersetDataTable div div').text()).toEqual('No rows found');
  });
});
