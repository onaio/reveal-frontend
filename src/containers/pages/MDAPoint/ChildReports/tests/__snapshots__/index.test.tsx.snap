// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MDA Reports/MDAPlansList should render school reports correctly: breadcrumbs 1`] = `
Array [
  <li
    className="breadcrumb-item"
  >
    <Link
      key="0"
      to="/"
    >
      <LinkAnchor
        href="/"
        navigate={[Function]}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </LinkAnchor>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="1"
      to="/intervention/mda-point/report"
    >
      <LinkAnchor
        href="/intervention/mda-point/report"
        navigate={[Function]}
      >
        <a
          href="/intervention/mda-point/report"
          onClick={[Function]}
        >
          MDA Point Plans
        </a>
      </LinkAnchor>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="2"
      to="/intervention/mda-point/report/40357eff-81b6-4e32-bd3d-484019689f7c"
    >
      <LinkAnchor
        href="/intervention/mda-point/report/40357eff-81b6-4e32-bd3d-484019689f7c"
        navigate={[Function]}
      >
        <a
          href="/intervention/mda-point/report/40357eff-81b6-4e32-bd3d-484019689f7c"
          onClick={[Function]}
        >
          MDA-Point-2019-09-05-TEST
        </a>
      </LinkAnchor>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="3"
      to="/intervention/mda-point/report/40357eff-81b6-4e32-bd3d-484019689f7c/2942"
    >
      <LinkAnchor
        href="/intervention/mda-point/report/40357eff-81b6-4e32-bd3d-484019689f7c/2942"
        navigate={[Function]}
      >
        <a
          href="/intervention/mda-point/report/40357eff-81b6-4e32-bd3d-484019689f7c/2942"
          onClick={[Function]}
        >
          Lusaka
        </a>
      </LinkAnchor>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="4"
      to="/intervention/mda-point/report/40357eff-81b6-4e32-bd3d-484019689f7c/3019"
    >
      <LinkAnchor
        href="/intervention/mda-point/report/40357eff-81b6-4e32-bd3d-484019689f7c/3019"
        navigate={[Function]}
      >
        <a
          href="/intervention/mda-point/report/40357eff-81b6-4e32-bd3d-484019689f7c/3019"
          onClick={[Function]}
        >
          Mtendere
        </a>
      </LinkAnchor>
    </Link>
  </li>,
  <li
    aria-current="page"
    className="active breadcrumb-item"
  >
    Akros_1
  </li>,
]
`;

exports[`components/MDA Reports/MDAPlansList should render school reports correctly: table 1`] = `
<table
  className="table table-striped table-bordered plans-list"
>
  <thead
    className="listview-thead"
  >
    <tr
      className="listview-tr"
    >
      <ElementMap
        HTMLTag="th"
        className="listview-td"
        items={
          Array [
            "Child's name",
            "National ID",
            "Enrolled in school",
            "MMA drug administered",
            "SACs Refused",
            "SACs Sick/Pregnant/Contraindicated",
            "ADR Reported",
            "PZQ tablets distributed",
            "Alb Tablets Distributed",
          ]
        }
      >
        <th
          className="listview-td"
          key="0"
        >
          Child's name
        </th>
        <th
          className="listview-td"
          key="1"
        >
          National ID
        </th>
        <th
          className="listview-td"
          key="2"
        >
          Enrolled in school
        </th>
        <th
          className="listview-td"
          key="3"
        >
          MMA drug administered
        </th>
        <th
          className="listview-td"
          key="4"
        >
          SACs Refused
        </th>
        <th
          className="listview-td"
          key="5"
        >
          SACs Sick/Pregnant/Contraindicated
        </th>
        <th
          className="listview-td"
          key="6"
        >
          ADR Reported
        </th>
        <th
          className="listview-td"
          key="7"
        >
          PZQ tablets distributed
        </th>
        <th
          className="listview-td"
          key="8"
        >
          Alb Tablets Distributed
        </th>
      </ElementMap>
    </tr>
  </thead>
  <tbody
    className="listview-tbody"
  >
    <tr
      className="listview-tr"
      key="0"
    >
      <ElementMap
        HTMLTag="td"
        className="listview-th"
        items={
          Array [
            "annonymous user",
            "2345",
            "No",
            "Yes",
            "No",
            "Yes",
            "No",
            0,
            0,
          ]
        }
      >
        <td
          className="listview-th"
          key="0"
        >
          annonymous user
        </td>
        <td
          className="listview-th"
          key="1"
        >
          2345
        </td>
        <td
          className="listview-th"
          key="2"
        >
          No
        </td>
        <td
          className="listview-th"
          key="3"
        >
          Yes
        </td>
        <td
          className="listview-th"
          key="4"
        >
          No
        </td>
        <td
          className="listview-th"
          key="5"
        >
          Yes
        </td>
        <td
          className="listview-th"
          key="6"
        >
          No
        </td>
        <td
          className="listview-th"
          key="7"
        >
          0
        </td>
        <td
          className="listview-th"
          key="8"
        >
          0
        </td>
      </ElementMap>
    </tr>
    <tr
      className="listview-tr"
      key="1"
    >
      <ElementMap
        HTMLTag="td"
        className="listview-th"
        items={
          Array [
            "known man",
            "843",
            "Yes",
            "No",
            "Yes",
            "No",
            "No",
            0,
            0,
          ]
        }
      >
        <td
          className="listview-th"
          key="0"
        >
          known man
        </td>
        <td
          className="listview-th"
          key="1"
        >
          843
        </td>
        <td
          className="listview-th"
          key="2"
        >
          Yes
        </td>
        <td
          className="listview-th"
          key="3"
        >
          No
        </td>
        <td
          className="listview-th"
          key="4"
        >
          Yes
        </td>
        <td
          className="listview-th"
          key="5"
        >
          No
        </td>
        <td
          className="listview-th"
          key="6"
        >
          No
        </td>
        <td
          className="listview-th"
          key="7"
        >
          0
        </td>
        <td
          className="listview-th"
          key="8"
        >
          0
        </td>
      </ElementMap>
    </tr>
  </tbody>
</table>
`;
